alg/gdalwarpkernel.cpp

```diff
@@ -5068,6 +5068,185 @@ GWKCheckAndComputeSrcOffsets(GWKJobStruct *psJob, int *_pabSuccess, int _iDstX,
     return true;
 }
 
+/************************************************************************/
+/*                   GWKOneSourceCornerFailsToReproject()               */
+/************************************************************************/
+
+static bool GWKOneSourceCornerFailsToReproject(GWKJobStruct *psJob)
+{
+    GDALWarpKernel *poWK = psJob->poWK;
+    for (int iY = 0; iY <= 1; ++iY)
+    {
+        for (int iX = 0; iX <= 1; ++iX)
+        {
+            double dfXTmp = poWK->nSrcXOff + iX * poWK->nSrcXSize;
```

Comment:
```suggestion
            const double dfXTmp = poWK->nSrcXOff + iX * poWK->nSrcXSize;
```

Reply:
it cannot be const, as it is updated by poWK->pfnTransformer()